src/device.rs

```diff
@@ -335,6 +344,11 @@ impl Device {
         Self::all_iter().find(|d| unique_id == d.unique_id())
     }
 
+    /// Returns the device matching the Vendor.
+    pub fn by_vendor(vendor_id: Vendor) -> Option<&'static Device> {
+        Self::all_iter().find(|d| vendor_id == d.vendor())
```

Comment:
I've missed that on the first review. I think `by_vendor` should return all matching devices and not only the first one, so it should be `filter()` instead of `find()`.

For the example it's fine if you then just pick the first device, it doesn't need to run on all.

Reply:
Thanks for the review, yes I have updated in my latest commit. Please take a look.